msteams-platform/m365-apps/combine-office-add-in-and-teams-app.md

```diff
@@ -0,0 +1,621 @@
+    |-- appPackage/
+    |-- build
+    |-- env/
+    |-- infra/
+    |-- tab/
+    |-- |-- infra/
+    |-- |-- node_modules/
+    |-- |-- src/
+    |-- |-- .webappignore
+    |-- |-- package-lock.json
+    |-- |-- package.json
+    |-- |-- tsconfig.json
+    |-- |-- web.config
+    |-- .gitignore
+    |-- .localConfigs
+    |-- teamsapp.local.yml
+    |-- teamsapp.yml
+    ```
+
+1. In the package.json that you just moved to the tab folder, delete the script named "dev:teamsfx" from the "scripts" object. This script is added to a new package.json in the next step.
+1. Create a new file named package.json *in the root of the project* and give it the following content:
+
+    ```
+    {
+        "name": "CombinedTabAndAddin",
+        "version": "0.0.1",
+        "author": "Contoso",
+        "scripts": {
+            "dev:teamsfx": "env-cmd --silent -f .localConfigs npm run start:tab",
+            "build:tab": "cd tab && npm run build",
+            "install:tab": "cd tab && npm install",
+            "start:tab": "cd tab && npm run start",
+            "test": "echo \"Error: no test specified\" && exit 1"
+        },
+        "devDependencies": {
+            "@microsoft/teamsfx-cli": "2.0.2-alpha.4f379e6ab.0",
+            "@microsoft/teamsfx-run-utils": "alpha",
+            "env-cmd": "^10.1.0",
+            "office-addin-dev-settings": "^2.0.3",
+            "ncp": "^2.0.0"
+        }
+    }
+    ```
+
+1. Change the "name", "version", and "author" properties, as needed.
+1. Open the teamsapp.yml file in the root of the project and find the line `ignoreFile: .webappignore`. Change this to `ignoreFile: ./tab/.webappignore`.
+1. Open the teamsapp.local.yml file in the root of the project and find the line `args: install --no-audit`. Change this to `args: run install:tab --no-audit`.
+1. Open **TERMINAL** in Visual Studio Code. Navigate to the root of the project and run `npm install`. Among other things, a new node_modules folder and a new package.lock.json file are created in the project root. 
+1. Next run `npm run install:tab`. Among other things, a new node_modules folder and a new package.lock.json file are created are created in the tab folder, if they aren't there already. 
+1. Verify that you can sideload the tab with the following steps:
+
+    1. Open Teams Toolkit, and in the **ACCOUNTS** section, verify that you're signed into Microsoft 365.
+    1. Select **View** | **Run** in Visual Studio Code.
+    1. In the **RUN AND DEBUG** drop down menu, select the option, **Debug in Teams (Edge)**, and then press F5. The project builds and runs. This process can take a couple of minutes. Eventually, Teams opens in a browser with a prompt to add your tab app.
+
+      > [!Note]
+      > If this is the first time you've debugged a Teams app on this computer, you're prompted to install an SSL certificate. Select **Install** and then **Yes** to the second prompt. Login to your Microsoft 365 account if you're prompted to do so.
```

Comment:
```suggestion
      > If this is the first time you've debugged a Teams app on this computer, you're prompted to install an SSL certificate. Select **Install** > **Yes**. Login to your Microsoft 365 account if you're prompted to do so.
```

Reply:
> 

The proposed version will not be as easy for the user. They need to know that a second prompt is expected.

Reply:
Agreed. Thanks.